File: gluecodium/src/test/resources/smoke/escaped_names/output/android/com/example/package/Class.java
/*
 *
 */
package com.example.package;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;
public final class Class extends NativeBase implements Interface {
    public Class() {
        this(constructor(), (Object)null);
        cacheThisInstance();
    }
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected Class(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    private native void cacheThisInstance();
    private static native long constructor();
    @NonNull
    public native Types.Struct fun(@NonNull final List<Types.Struct> double) throws Types.ExceptionException;
    @NonNull
    public native Types.Enum getProperty();
    public native void setProperty(@NonNull final Types.Enum value);
}

File: gluecodium/src/test/resources/smoke/escaped_names/output/android-kotlin/com/example/package/Class.kt
/*

 *
 */

@file:JvmName("classExtensions")

package com.example.package

import com.example.NativeBase

class Class : NativeBase, Interface {




    constructor() : this(Constructor(), null as Any?) {
        cacheThisInstance();
    }

    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}

    private external fun cacheThisInstance()



    @Throws(Types.ExceptionException::class)
    external fun Fun(double: MutableList<Types.Struct>) : Types.Struct

    var Property: Types.Enum
        external get
        external set




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

        @JvmStatic external fun Constructor() : Long
    }
}

